api/graphql/types/rejected_occurrence/filtersets.py

```diff
@@ -0,0 +1,100 @@
+import django_filters
+from django.contrib.postgres.search import SearchVector
+from django.db import models
+from graphene_django_extensions import ModelFilterSet
+from graphene_django_extensions.filters import IntChoiceFilter, IntMultipleChoiceFilter
+from lookup_property import L
+
+from common.db import raw_prefixed_query
+from reservations.models import RejectedOccurrence
+
+__all__ = [
+    "RejectedOccurrenceFilterSet",
+]
+
+from reservations.querysets import RejectedOccurrenceQuerySet
+
+
+class RejectedOccurrenceFilterSet(ModelFilterSet):
+    pk = IntMultipleChoiceFilter()
+    recurring_reservation = IntChoiceFilter()
+    application_round = IntChoiceFilter(
+        field_name=(
+            "recurring_reservation"
+            "__allocated_time_slot"
+            "__reservation_unit_option"
+            "__application_section"
+            "__application"
+            "__application_round"
+        )
+    )
+    reservation_unit = IntChoiceFilter(field_name="recurring_reservation__reservation_unit")
+    unit = IntChoiceFilter(field_name="recurring_reservation__reservation_unit__unit")
+    text_search = django_filters.CharFilter(method="filter_text_search")
+
+    class Meta:
+        model = RejectedOccurrence
+        order_by = [
```

Comment:
Let's add `rejection_reason` as well, it was missing from the Jira ticket.

Reply:
What should be the order when ordering by `rejection_reason`?

Reply:
Implemented ordering (asc):
- `INTERVAL_NOT_ALLOWED`
- `OVERLAPPING_RESERVATIONS`
- `RESERVATION_UNIT_CLOSED`